packages/flutter/lib/src/widgets/mimic.dart

```diff
@@ -19,6 +23,73 @@ class MimicableKey {
   }
 }
 
+class MimicOverlayEntry {
+  MimicOverlayEntry._(this._key) {
+    _overlayEntry = new OverlayEntry(builder: _build);
+  }
+
+  MimicableKey _key;
+  OverlayEntry _overlayEntry;
+
+  // Animation state
+  GlobalKey _targetKey;
+  Curve _curve;
+  Performance _performance;
+
+  Future animateTo({
+    GlobalKey targetKey,
+    Duration duration,
+    Curve curve: Curves.linear
+  }) {
+    assert(_key != null);
+    assert(_overlayEntry != null);
+    assert(targetKey != null);
+    assert(duration != null);
+    assert(curve != null);
+    _targetKey = targetKey;
+    _curve = curve;
+    // TODO(abarth): Support changing the animation target when in flight.
+    assert(_performance == null);
+    _performance = new Performance(duration: duration)
+      ..addListener(_overlayEntry.markNeedsBuild);
+    return _performance.play();
+  }
+
+  void stopMimic() {
+    _targetKey = null;
+    _curve = null;
+    _performance?.stop();
+    _performance = null;
+    _key.stopMimic();
+    _key = null;
+    _overlayEntry.remove();
+    _overlayEntry = null;
+  }
+
+  Widget _build(BuildContext context) {
+    assert(_key != null);
+    assert(_overlayEntry != null);
+    Rect bounds = _key.globalBounds;
+    if (_targetKey != null) {
+      assert(_performance != null);
+      assert(_curve != null);
+      RenderBox box = _targetKey.currentContext?.findRenderObject();
+      if (box != null) {
+        // TODO(abarth): Handle non-translation coordinate changes.
+        Rect targetBounds = box.localToGlobal(Point.origin) & box.size;
```

Comment:
The overlay might not be at global 0,0, you have to convert back down to the overlay coordinate space.